pkg/mgmterrors/aterrors.go

```diff
@@ -0,0 +1,183 @@
+/*
+ (c) Copyright [2021-2023] Micro Focus or one of its affiliates.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package mgmterrors
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/cloud"
+	"github.com/vertica/vertica-kubernetes/pkg/events"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type ATErrors struct {
+	Writer               EVWriter
+	VDB                  *vapi.VerticaDB
+	GenericFailureReason string // The failure reason when no specific error is found
+}
+
+func MakeATErrors(writer EVWriter, vdb *vapi.VerticaDB, genericFailureReason string) EventLogger {
+	return &ATErrors{
+		Writer:               writer,
+		VDB:                  vdb,
+		GenericFailureReason: genericFailureReason,
+	}
+}
+
+const (
+	// Amount of time to wait after a restart failed because nodes weren't down yet
+	RestartNodesNotDownRequeueWaitTimeInSeconds = 10
+)
+
+// LogFailureError is called when admintools had attempted an option but
```

Comment:
function's name is LogFailure.

Reply:
Good catch. This was an early name of this function. I forget to change this.